### ISSUE TITLE
The linker problem about LLVM

### ISSUE THREAD

**Issue by holyprince** [ROCm/HIP#979]
When I using hipcc to link some ported function from cuda. The compile process now doesn't have an error.

But the linker shows the error info:

```
LLVM ERROR: Cannot select: 0x44b01b8: i1 = add 0x44b0f20, Constant:i1<-1>
  0x44b0f20: i1,ch = CopyFromReg 0x42dfaa8, Register:i1 %28
    0x44ae610: i1 = Register %28
  0x48da798: i1 = Constant<-1>
In function: _Z23hip_kernel_diff2_coarseILb0ELb0ELi512ELi1ELi2EEvPfS0_S0_S0_S0_S0_18AccProjectorKernelS0_S0_ii
Generating AMD GCN kernel failed in llc for target: gfx906
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

I also have checked the AccProjectorKernel (following next ) and try to define the flag PROJECTOR_NO_TEXTURES to using the global memory but the error is the same.
I also try to change the  AMDGPU_TARGET by
export HCC_AMDGPU_TARGET=gfx900 also doesn't work. #141 

```c++
#ifndef PROJECTOR_NO_TEXTURES
#define PROJECTOR_PTR_TYPE hipTextureObject_t
#else
#define PROJECTOR_PTR_TYPE XFLOAT *
#endif
class AccProjectorKernel
{

public:
	int mdlX, mdlXY, mdlZ,
		imgX, imgY, imgZ,
		mdlInitY, mdlInitZ,
		padding_factor,
		maxR, maxR2;

	PROJECTOR_PTR_TYPE mdlReal;
	PROJECTOR_PTR_TYPE mdlImag;
#ifdef HIP
	PROJECTOR_PTR_TYPE mdlComplex;
#else
	std::complex<XFLOAT> *mdlComplex;
#endif

	AccProjectorKernel(
			int mdlX, int mdlY, int mdlZ,
			int imgX, int imgY, int imgZ,
			int mdlInitY, int mdlInitZ,
			int padding_factor,
			int maxR,
#ifdef HIP
			PROJECTOR_PTR_TYPE mdlComplex
#else
			std::complex<XFLOAT> *mdlComplex
#endif
			):
			mdlX(mdlX), mdlXY(mdlX*mdlY), mdlZ(mdlZ),
			imgX(imgX), imgY(imgY), imgZ(imgZ),
			mdlInitY(mdlInitY), mdlInitZ(mdlInitZ),
			padding_factor(padding_factor),
			maxR(maxR), maxR2(maxR*maxR),
			mdlComplex(mdlComplex)
		{};

	AccProjectorKernel(
			int mdlX, int mdlY, int mdlZ,
			int imgX, int imgY, int imgZ,
			int mdlInitY, int mdlInitZ,
			int padding_factor,
			int maxR,
			PROJECTOR_PTR_TYPE mdlReal, PROJECTOR_PTR_TYPE mdlImag
			):
				mdlX(mdlX), mdlXY(mdlX*mdlY), mdlZ(mdlZ),
				imgX(imgX), imgY(imgY), imgZ(imgZ),
				mdlInitY(mdlInitY), mdlInitZ(mdlInitZ),
				padding_factor(padding_factor),
				maxR(maxR), maxR2(maxR*maxR),
				mdlReal(mdlReal), mdlImag(mdlImag)
			{
#ifndef HIP		
				std::complex<XFLOAT> *pData = mdlComplex;
				for(size_t i=0; i<(size_t)mdlX * (size_t)mdlY * (size_t)mdlZ; i++) {
					std::complex<XFLOAT> arrayval(*mdlReal ++, *mdlImag ++);
					pData[i] = arrayval;		        
				}
#endif
			};

#ifdef HIP
	__device__ __forceinline__
#endif
	void project3Dmodel(....)
        other function .....
}
```

**Comment by nartmada** [comment]
@holyprince, apologies for the lack of response from AMD.  Do you still need assistance with this issue?  Thanks.
